File: backend/src/main/java/vn/edu/fpt/doghandbook/backend/service/NutritionService.java
package vn.edu.fpt.doghandbook.backend.service;

import vn.edu.fpt.doghandbook.backend.dto.request.NutritionStandardRequest;
import vn.edu.fpt.doghandbook.backend.dto.response.PageResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.NutritionStandardResponse;

import java.util.List;

public interface NutritionService {

    PageResponse<NutritionStandardResponse> getAll(int page, int size, String search);

    NutritionStandardResponse getById(Integer id);

    List<NutritionStandardResponse> getByBreedId(Integer breedId);

    NutritionStandardResponse create(NutritionStandardRequest request, Integer createdByUserId);

    NutritionStandardResponse update(Integer id, NutritionStandardRequest request);

    void delete(Integer id);
}

File: web-admin/src/services/nutritionService.js
import api from './api';

export const nutritionService = {
  getAll: (page = 0, size = 10, search = '') =>
    api.get(`/nutrition-standards?page=${page}&size=${size}&search=${search}&sort=updatedAt,desc&sort=createdAt,desc`),
  getById: (id) => api.get(`/nutrition-standards/${id}`),
  create: (data) => api.post('/nutrition-standards', data),
  update: (id, data) => api.put(`/nutrition-standards/${id}`, data),
  delete: (id) => api.delete(`/nutrition-standards/${id}`),
  calculate: (data) => api.post('/nutrition/calculate', data),
};
